src/libmongoc/tests/test-conveniences.c

```diff
@@ -1302,6 +1302,16 @@ match_bson_arrays (const bson_t *array, const bson_t *pattern, match_ctx_t *ctx)
 
       derive (ctx, &derived, bson_iter_key (&array_iter));
 
+      if (ctx && ctx->visitor_fn) {
+         match_action_t action =
+            ctx->visitor_fn (ctx, &pattern_iter, &array_iter);
+         if (action == MATCH_ACTION_ABORT) {
+            return false;
+         } else if (action == MATCH_ACTION_SKIP) {
```

Comment:
What is the distinction between `MATCH_ACTION_SKIP` and `MATCH_ACTION_CONTINUE`? Initial read suggests to me (perhaps incorrectly) that `SKIP` should break out of the loop vs. `CONTINUE` to continue iteration.

Reply:
`MATCH_ACTION_SKIP` means the visitor wants the caller to skip any additional matching of this field.
`MATCH_ACTION_CONTINUE` means the visitor wants the caller proceed matching this field.
Added comments to clarify the handling of `action`.